internal/status/cache.go

```diff
@@ -38,22 +38,50 @@ const (
 func NewCache() Cache {
 	return Cache{
 		proxyUpdates: make(map[types.NamespacedName]*ProxyUpdate),
+		entries:      make(map[string]map[types.NamespacedName]CacheEntry),
 	}
 }
 
+type CacheEntry interface {
+	AsStatusUpdate() k8s.StatusUpdate
+	ConditionFor(ConditionType) *contour_api_v1.DetailedCondition
+}
+
 // Cache holds status updates from the DAG back towards Kubernetes.
 // It holds a per-Kind cache, and is intended to be accessed with a
 // KindAccessor.
 type Cache struct {
 	proxyUpdates map[types.NamespacedName]*ProxyUpdate
+
+	entries map[string]map[types.NamespacedName]CacheEntry
```

Comment:
it'd be helpful to reiterate in a comment here that the outer map is keyed on Kind.

Reply:
fixed